js/rpg_core/ImageCache.js

```diff
@@ -0,0 +1,101 @@
+function ImageCache(){
+    this.initialize.apply(this, arguments);
+}
+
+ImageCache.limit = 20 * 1000 * 1000;
+
+ImageCache.prototype.initialize = function(){
+    this._items = {};
+
+    //it holds purged bitmap which is not in ready.
+    this._purged = [];
+};
+
+ImageCache.prototype.add = function(key, value){
+    this._items[key] = {
+        bitmap: value,
+        touch: Date.now(),
+        key: key
+    };
+
+    this._truncateCache();
+};
+
+ImageCache.prototype.get = function(key){
+    if(this._items[key]){
+        var item = this._items[key];
+        item.touch = Date.now();
+        return item.bitmap;
+    }
+
+    return null;
+};
+
+ImageCache.prototype.reserve = function(key, reservationId){
+    this._items[key].reservationId = reservationId;
+};
+
+ImageCache.prototype.releaseReservation = function(reservationId){
+    var items = this._items;
+
+    Object.keys(items)
+        .map(function(key){return items[key];})
+        .forEach(function(item){
+            if(item.reservationId === reservationId){
+                delete item.reservationId;
+            }
+        });
+};
+
+ImageCache.prototype._truncateCache = function(){
+    var items = this._items;
+    var sizeLeft = ImageCache.limit;
+    var purged = this._purged;
+
+    Object.keys(items).map(function(key){
+        return items[key];
+    }).sort(function(a, b){
+        return b.touch - a.touch;
+    }).forEach(function(item){
+        if(sizeLeft > 0 || item.reservationId){
```

Comment:
Well, bitmap's size is below 3 patterns.

- loaded: width * height
- none: 1
- otherwise: 0

Therefore, non-loaded bitmaps don't squeeze `ImageCache.limit`.
So if non-loaded bitmaps are never purged, `ImageManager.isReady()` regains its original meaning.